core/chains/evm/logpoller/parser_test.go

```diff
@@ -0,0 +1,339 @@
+package logpoller
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+)
+
+func assertArgs(t *testing.T, args *queryArgs, numVals int) {
+	values, err := args.toArgs()
+
+	assert.Len(t, values, numVals)
+	assert.NoError(t, err)
+}
+
+func TestDSLParser(t *testing.T) {
```

Comment:
Probably all existing LP queries are worth testing to verify if the SQL generated by the DSL is the same as the original one. (and probably if the results returned from the db are the same)

Reply:
The current querying structure in this PR doesn't intend to satisfy all LP queries. There are some tests on some LP queries to ensure we are on the right track though.

Reply:
@EasterTheBunny Which LP queries are we unable to support? Will confidence levels and SigsExcluded support enable us to replicate every LP query?

Reply:
// do not include address or event hash; not supported by QueryKey at this time
// these can be tested at the ORM level to verify that the query structure could theoretically be used
SelectBlockByHash
SelectBlockByNumber
SelectLatestBlock
SelectOldestBlock
SelectLogsByBlockRange
GetBlocksRange

// expected support by QueryKey and can include in tests
SelectLogsCreatedAfter (tested)
SelectLogsDataWordBetween (tested)
SelectIndexedLogsCreatedAfter (tested)
SelectLogsWithSigs (tested)
SelectLatestLogByEventSigWithConfs
SelectLogs
SelectLogsDataWordRange
SelectLogsDataWordGreaterThan
SelectIndexedLogsTopicGreaterThan
SelectIndexedLogsTopicRange
SelectIndexedLogs
SelectIndexedLogsByBlockRange
SelectIndexedLogsByTxHash

// have specific pgsql queries and aren't currently supported
SelectLatestLogEventSigsAddrsWithConfs
SelectLatestBlockByEventSigsAddrsWithConfs
SelectIndexedLogsWithSigsExcluding

I can add more tests for the queries that are directly supported by QueryKey to ensure our test coverage is as high as possible.

Reply:
I added new tests to verify all queries supported by `QueryKey`.

Reply:
Ah ok, I completely forgot about non log related functions